lib/conversations_client.ex

```diff
@@ -76,16 +85,31 @@ defmodule ExMicrosoftBot.Client.Conversations do
   @doc """
   This method allows you to upload an attachment directly into a channels blob storage. [API Reference](https://docs.botframework.com/en-us/restapi/connector/#!/Conversations/Conversations_UploadAttachment)
   """
-  @spec upload_attachment(String.t, String.t, Models.AttachmentData.t) :: :ok | Client.error_type
+  @spec upload_attachment(String.t(), String.t(), Models.AttachmentData.t()) ::
+          {:ok, Models.ResourceResponse.t()} | Client.error_type()
   def upload_attachment(service_url, conversation_id, %Models.AttachmentData{} = attachment) do
     api_endpoint = "#{conversations_endpoint(service_url)}/#{conversation_id}/attachments"
 
-    HTTPotion.post(api_endpoint, authed_req_options(api_endpoint, body: Poison.encode!(attachment)))
+    HTTPotion.post(
+      api_endpoint,
+      authed_req_options(api_endpoint, body: Poison.encode!(attachment))
+    )
     |> deserialize_response(&Models.ResourceResponse.parse/1)
   end
 
-  defp change_deserialized_response_to_ok({:ok, _}), do: :ok
-  defp change_deserialized_response_to_ok(resp), do: resp
+  @doc """
+  Updates an existing activity. The activity struct is expected to have an ID.
```

Comment:
If there is a link to some API reference documentation for it can you add it here